azext_iot/digitaltwins/providers/connection/builders.py

```diff
@@ -0,0 +1,290 @@
+        eh_entity_path: str,
+        eh_resource_group: str,
+        eh_subscription: str,
+        eh_consumer_group: str = '$Default',
+        yes: bool = False,
+    ):
+        self.cli = EmbeddedCLI()
+        self.yes = yes
+        self.error_prefix = "Could not create ADT instance connection. Unable to"
+        self.dt = dt_instance
+        if self.dt.identity is None:
+            raise CLIError(
+                "DT instance does not have System-Assigned Identity enabled. Please enable and try again."
+            )
+
+        # Populate adx_cluster_uri, adx_location, adx_resource_id and perform checks
+        self.validate_adx(
+            adx_cluster_name=adx_cluster_name,
+            adx_database_name=adx_database_name,
+            adx_resource_group=adx_resource_group,
+            adx_subscription=adx_subscription,
+        )
+
+        self.eh_consumer_group = eh_consumer_group
+        # Populate eh_endpoint_uri, eh_namespace_resource_id and perform checks
+        self.validate_eventhub(
+            eh_namespace=eh_namespace,
+            eh_entity_path=eh_entity_path,
+            eh_resource_group=eh_resource_group,
+            eh_subscription=eh_subscription,
+            eh_consumer_group=eh_consumer_group,
+        )
+
+    def validate_eventhub(
+        self,
+        eh_namespace: str,
+        eh_entity_path: str,
+        eh_resource_group: str,
+        eh_subscription: str,
+        eh_consumer_group: str
+    ):
+        from azext_iot.digitaltwins.providers.endpoint.builders import EventHubEndpointBuilder
+        eh_endpoint = EventHubEndpointBuilder(
+            endpoint_resource_name=eh_entity_path,
+            endpoint_resource_group=eh_resource_group,
+            endpoint_resource_namespace=eh_namespace,
+            endpoint_resource_policy=None,
+            endpoint_subscription=eh_subscription,
+        )
+        eh_endpoint.error_prefix = self.error_prefix + " find"
+        self.eh_endpoint_uri = eh_endpoint.build_identity_based().endpoint_uri
+
+        eh_consumer_group_op = self.cli.invoke(
+            "eventhubs eventhub consumer-group show -n {} --eventhub-name {} --namespace-name {} -g {}".format(
+                eh_consumer_group,
+                eh_entity_path,
+                eh_namespace,
+                eh_resource_group,
+            ),
+            subscription=eh_subscription,
+        )
+        if not eh_consumer_group_op.success():
+            raise CLIError("{} retrieve Event Hub Consumer Group.".format(self.error_prefix))
+
+        self.eh_namespace_resource_id = (
+            "/subscriptions/{}/resourceGroups/{}/providers/Microsoft.EventHub/namespaces/{}".format(
+                eh_subscription,
+                eh_resource_group,
+                eh_namespace,
+            )
+        )
+        logger.debug("Finished checking the EventHub resource.")
+
+        self.add_dt_role_assignment(
+            role="Azure Event Hubs Data Owner",
```

Comment:
Let's extract rbac role strings in a common area to be referenced....like the dt common.py module or elsewhere with similar intent.